vaadin-radio-button-flow-demo/src/main/java/com/vaadin/flow/component/radiobutton/demo/RadioButtonGroupView.java

```diff
@@ -15,330 +15,287 @@
+            } else {
+                value.setText("Selected: " + event.getValue());
+            }
+        });
         // end-source-example
 
-        group.setId("button-group-disabled");
-
-        addCard("Disabled radio button group", group);
+        addCard("Value change event", radioGroup, value);
     }
 
-    private void addReadOnlyGroup() {
+    private void configurationForReqired() {
         // begin-source-example
-        // source-example-heading: Read-only radio button group
-        Div valueInfo = new Div();
-
-        RadioButtonGroup<String> group = new RadioButtonGroup<>();
-        group.setItems("foo", "bar", "baz");
-        group.setReadOnly(true);
-
-        NativeButton button = new NativeButton("Switch read-only state",
-                event -> group.setReadOnly(!group.isReadOnly()));
-        group.addValueChangeListener(
-                event -> valueInfo.setText(group.getValue()));
+        // source-example-heading: Required
+        Employee employee = new Employee();
+        Binder<Employee> binder = new Binder<>();
+
+        RadioButtonGroup<String> radioGroup = new RadioButtonGroup<>();
+        radioGroup.setLabel("Department");
+        radioGroup.setItems("Account Manager", "Designer", "Marketing Manager",
+                "Developer");
+        radioGroup.addThemeVariants(RadioGroupVariant.LUMO_VERTICAL);
+
+        binder.forField(radioGroup).asRequired("Please choose a department")
+                .bind(Employee::getTitle, Employee::setTitle);
+
+        Button button = new Button("Submit", event -> {
+            if (binder.writeBeanIfValid(employee)) {
+                Notification.show("Submit successful", 2000,
+                        Notification.Position.MIDDLE);
+            }
+        });
         // end-source-example
 
-        group.setId("button-group-read-only");
-        valueInfo.setId("selected-value-info");
-        button.setId("switch-read-only");
+        radioGroup.getStyle().set("margin-right", "5.5em");
+        HorizontalLayout layout = new HorizontalLayout(radioGroup, button);
+        layout.setAlignItems(FlexComponent.Alignment.BASELINE);
+        addCard("Validation", "Required", layout);
 
-        addCard("Read-only radio button group", group, button, valueInfo);
     }
 
-    private void addDisabledItems() {
-
-        Div valueInfo = new Div();
+    private void customOptions() {
         // begin-source-example
-        // source-example-heading: Radio button group with item enabled provider
-        RadioButtonGroup<String> group = new RadioButtonGroup<>();
-        group.setItems("foo", "bar", "baz");
-        group.setItemEnabledProvider(item -> !"bar".equals(item));
+        // source-example-heading: Customizing radio button label
+        RadioButtonGroup<Information> radioButton = new RadioButtonGroup<>();
+        radioButton.setLabel("User");
+        radioButton.setItems(
+                new Information("Gabriella",
```

Comment:
"Information" class used here seems to wrap the name and an image of a person. One of the other demos used "Employee" class, which also represents a person, so it would make sense to combine the properties from "Information" to "Employee" and use the same "Employee" class in both demos.

Reply:
Done.